lib/rules/selector-nested-pattern/index.cjs

```diff
@@ -53,21 +56,41 @@ const rule = (primary, secondaryOptions) => {
 				return;
 			}
 
-			const selectors = splitList ? ruleNode.selectors : [ruleNode.selector];
+			if (splitList) {
+				parseSelector(getRuleSelector(ruleNode), result, ruleNode)?.each((selector) => {
+					const { index, endIndex, selector: selectorStr } = getStrippedSelectorSource(selector);
 
-			for (const selector of selectors) {
-				if (normalizedPattern.test(selector)) {
-					continue;
-				}
+					if (normalizedPattern.test(selectorStr)) {
+						return;
+					}
 
-				report({
-					result,
-					ruleName,
-					message: messages.expected,
-					messageArgs: [selector, primary],
-					node: ruleNode,
-					word: selector,
+					report({
+						result,
+						ruleName,
+						message: messages.expected,
+						messageArgs: [selectorStr, primary],
+						node: ruleNode,
+						index,
+						endIndex,
+					});
 				});
+			} else {
+				const selectors = [ruleNode.selector];
+
+				for (const selector of selectors) {
```

Comment:
Now that there is differentiation between the two modes I think it is fine to optimize this is a bit.

Can you remove the for loop?

Reply:
Ah, you're right, good point. I've fixed it!
[chore: remove unnecessary for loop](https://github.com/stylelint/stylelint/pull/8072/commits/87bc953002cd75b0f16117f9d1fae8eed6c02a02)